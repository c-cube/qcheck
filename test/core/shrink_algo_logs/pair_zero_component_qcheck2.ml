open QCheck2

let pair_one_zero =
  Test.make ~name:"pairs have a zero component"
    Gen.(pair int int)
    (Log.shrinks Print.(pair int int) @@
     fun (i,j) -> i=0 || j=0)
;;
try Test.check_exn pair_one_zero with Test.Test_fail _ -> ()
