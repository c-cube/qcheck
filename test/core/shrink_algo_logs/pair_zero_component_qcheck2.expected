fails (-4571327332697646483, -3285013039971199785)
fails (2732179349098564728, -3285013039971199785)
holds (0, -3285013039971199785)
fails (1366089674549282364, -3285013039971199785)
holds (0, -3285013039971199785)
fails (683044837274641182, -3285013039971199785)
holds (0, -3285013039971199785)
fails (341522418637320591, -3285013039971199785)
holds (0, -3285013039971199785)
fails (170761209318660295, -3285013039971199785)
holds (0, -3285013039971199785)
fails (85380604659330147, -3285013039971199785)
holds (0, -3285013039971199785)
fails (42690302329665073, -3285013039971199785)
holds (0, -3285013039971199785)
fails (21345151164832536, -3285013039971199785)
holds (0, -3285013039971199785)
fails (10672575582416268, -3285013039971199785)
holds (0, -3285013039971199785)
fails (5336287791208134, -3285013039971199785)
holds (0, -3285013039971199785)
fails (2668143895604067, -3285013039971199785)
holds (0, -3285013039971199785)
fails (1334071947802033, -3285013039971199785)
holds (0, -3285013039971199785)
fails (667035973901016, -3285013039971199785)
holds (0, -3285013039971199785)
fails (333517986950508, -3285013039971199785)
holds (0, -3285013039971199785)
fails (166758993475254, -3285013039971199785)
holds (0, -3285013039971199785)
fails (83379496737627, -3285013039971199785)
holds (0, -3285013039971199785)
fails (41689748368813, -3285013039971199785)
holds (0, -3285013039971199785)
fails (20844874184406, -3285013039971199785)
holds (0, -3285013039971199785)
fails (10422437092203, -3285013039971199785)
holds (0, -3285013039971199785)
fails (5211218546101, -3285013039971199785)
holds (0, -3285013039971199785)
fails (2605609273050, -3285013039971199785)
holds (0, -3285013039971199785)
fails (1302804636525, -3285013039971199785)
holds (0, -3285013039971199785)
fails (651402318262, -3285013039971199785)
holds (0, -3285013039971199785)
fails (325701159131, -3285013039971199785)
holds (0, -3285013039971199785)
fails (162850579565, -3285013039971199785)
holds (0, -3285013039971199785)
fails (81425289782, -3285013039971199785)
holds (0, -3285013039971199785)
fails (40712644891, -3285013039971199785)
holds (0, -3285013039971199785)
fails (20356322445, -3285013039971199785)
holds (0, -3285013039971199785)
fails (10178161222, -3285013039971199785)
holds (0, -3285013039971199785)
fails (5089080611, -3285013039971199785)
holds (0, -3285013039971199785)
fails (2544540305, -3285013039971199785)
holds (0, -3285013039971199785)
fails (1272270152, -3285013039971199785)
holds (0, -3285013039971199785)
fails (636135076, -3285013039971199785)
holds (0, -3285013039971199785)
fails (318067538, -3285013039971199785)
holds (0, -3285013039971199785)
fails (159033769, -3285013039971199785)
holds (0, -3285013039971199785)
fails (79516884, -3285013039971199785)
holds (0, -3285013039971199785)
fails (39758442, -3285013039971199785)
holds (0, -3285013039971199785)
fails (19879221, -3285013039971199785)
holds (0, -3285013039971199785)
fails (9939610, -3285013039971199785)
holds (0, -3285013039971199785)
fails (4969805, -3285013039971199785)
holds (0, -3285013039971199785)
fails (2484902, -3285013039971199785)
holds (0, -3285013039971199785)
fails (1242451, -3285013039971199785)
holds (0, -3285013039971199785)
fails (621225, -3285013039971199785)
holds (0, -3285013039971199785)
fails (310612, -3285013039971199785)
holds (0, -3285013039971199785)
fails (155306, -3285013039971199785)
holds (0, -3285013039971199785)
fails (77653, -3285013039971199785)
holds (0, -3285013039971199785)
fails (38826, -3285013039971199785)
holds (0, -3285013039971199785)
fails (19413, -3285013039971199785)
holds (0, -3285013039971199785)
fails (9706, -3285013039971199785)
holds (0, -3285013039971199785)
fails (4853, -3285013039971199785)
holds (0, -3285013039971199785)
fails (2426, -3285013039971199785)
holds (0, -3285013039971199785)
fails (1213, -3285013039971199785)
holds (0, -3285013039971199785)
fails (606, -3285013039971199785)
holds (0, -3285013039971199785)
fails (303, -3285013039971199785)
holds (0, -3285013039971199785)
fails (151, -3285013039971199785)
holds (0, -3285013039971199785)
fails (75, -3285013039971199785)
holds (0, -3285013039971199785)
fails (37, -3285013039971199785)
holds (0, -3285013039971199785)
fails (18, -3285013039971199785)
holds (0, -3285013039971199785)
fails (9, -3285013039971199785)
holds (0, -3285013039971199785)
fails (4, -3285013039971199785)
holds (0, -3285013039971199785)
fails (2, -3285013039971199785)
holds (0, -3285013039971199785)
fails (1, -3285013039971199785)
holds (0, -3285013039971199785)
fails (1, 1281206205950578423)
holds (0, 1281206205950578423)
holds (1, 0)
fails (1, 640603102975289211)
holds (0, 640603102975289211)
holds (1, 0)
fails (1, 320301551487644605)
holds (0, 320301551487644605)
holds (1, 0)
fails (1, 160150775743822302)
holds (0, 160150775743822302)
holds (1, 0)
fails (1, 80075387871911151)
holds (0, 80075387871911151)
holds (1, 0)
fails (1, 40037693935955575)
holds (0, 40037693935955575)
holds (1, 0)
fails (1, 20018846967977787)
holds (0, 20018846967977787)
holds (1, 0)
fails (1, 10009423483988893)
holds (0, 10009423483988893)
holds (1, 0)
fails (1, 5004711741994446)
holds (0, 5004711741994446)
holds (1, 0)
fails (1, 2502355870997223)
holds (0, 2502355870997223)
holds (1, 0)
fails (1, 1251177935498611)
holds (0, 1251177935498611)
holds (1, 0)
fails (1, 625588967749305)
holds (0, 625588967749305)
holds (1, 0)
fails (1, 312794483874652)
holds (0, 312794483874652)
holds (1, 0)
fails (1, 156397241937326)
holds (0, 156397241937326)
holds (1, 0)
fails (1, 78198620968663)
holds (0, 78198620968663)
holds (1, 0)
fails (1, 39099310484331)
holds (0, 39099310484331)
holds (1, 0)
fails (1, 19549655242165)
holds (0, 19549655242165)
holds (1, 0)
fails (1, 9774827621082)
holds (0, 9774827621082)
holds (1, 0)
fails (1, 4887413810541)
holds (0, 4887413810541)
holds (1, 0)
fails (1, 2443706905270)
holds (0, 2443706905270)
holds (1, 0)
fails (1, 1221853452635)
holds (0, 1221853452635)
holds (1, 0)
fails (1, 610926726317)
holds (0, 610926726317)
holds (1, 0)
fails (1, 305463363158)
holds (0, 305463363158)
holds (1, 0)
fails (1, 152731681579)
holds (0, 152731681579)
holds (1, 0)
fails (1, 76365840789)
holds (0, 76365840789)
holds (1, 0)
fails (1, 38182920394)
holds (0, 38182920394)
holds (1, 0)
fails (1, 19091460197)
holds (0, 19091460197)
holds (1, 0)
fails (1, 9545730098)
holds (0, 9545730098)
holds (1, 0)
fails (1, 4772865049)
holds (0, 4772865049)
holds (1, 0)
fails (1, 2386432524)
holds (0, 2386432524)
holds (1, 0)
fails (1, 1193216262)
holds (0, 1193216262)
holds (1, 0)
fails (1, 596608131)
holds (0, 596608131)
holds (1, 0)
fails (1, 298304065)
holds (0, 298304065)
holds (1, 0)
fails (1, 149152032)
holds (0, 149152032)
holds (1, 0)
fails (1, 74576016)
holds (0, 74576016)
holds (1, 0)
fails (1, 37288008)
holds (0, 37288008)
holds (1, 0)
fails (1, 18644004)
holds (0, 18644004)
holds (1, 0)
fails (1, 9322002)
holds (0, 9322002)
holds (1, 0)
fails (1, 4661001)
holds (0, 4661001)
holds (1, 0)
fails (1, 2330500)
holds (0, 2330500)
holds (1, 0)
fails (1, 1165250)
holds (0, 1165250)
holds (1, 0)
fails (1, 582625)
holds (0, 582625)
holds (1, 0)
fails (1, 291312)
holds (0, 291312)
holds (1, 0)
fails (1, 145656)
holds (0, 145656)
holds (1, 0)
fails (1, 72828)
holds (0, 72828)
holds (1, 0)
fails (1, 36414)
holds (0, 36414)
holds (1, 0)
fails (1, 18207)
holds (0, 18207)
holds (1, 0)
fails (1, 9103)
holds (0, 9103)
holds (1, 0)
fails (1, 4551)
holds (0, 4551)
holds (1, 0)
fails (1, 2275)
holds (0, 2275)
holds (1, 0)
fails (1, 1137)
holds (0, 1137)
holds (1, 0)
fails (1, 568)
holds (0, 568)
holds (1, 0)
fails (1, 284)
holds (0, 284)
holds (1, 0)
fails (1, 142)
holds (0, 142)
holds (1, 0)
fails (1, 71)
holds (0, 71)
holds (1, 0)
fails (1, 35)
holds (0, 35)
holds (1, 0)
fails (1, 17)
holds (0, 17)
holds (1, 0)
fails (1, 8)
holds (0, 8)
holds (1, 0)
fails (1, 4)
holds (0, 4)
holds (1, 0)
fails (1, 2)
holds (0, 2)
holds (1, 0)
fails (1, 1)
holds (0, 1)
holds (1, 0)
